src/Compilers/VisualBasic/Portable/Symbols/BaseTypeAnalysis.vb

```diff
@@ -190,7 +190,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
 
             Debug.Assert(root.OriginalDefinition = root, "root must not be a substitution")
 
-            If current Is Nothing Then
+            If current Is Nothing Or TypeOf current Is MissingMetadataTypeSymbol Then
```

Comment:
@agocke would this be considered a declaration level error?  If so then I agree we should not get here.  If not then indeed the code can get this far.